### PR TITLE
HiveNavigation UI

### PR DESCRIPTION
**Pull request by hasparus**
I implemented the new Navigation Menu as designed with a nice "Stripe-landing-page" style transition.

On mobiles, we'll use the ordinary Nextra Navbar.
https://github.com/shuding/nextra/blob/b50e325ed47f059b11474fc4ed9aa433d29fb984/packages/nextra-theme-docs/src/components/navbar.tsx#L11

- [x] I don't like the opacity transition in the content, gotta spent a 20 more minutes on it to get it just right.
- [x] I'm assuming it's also used in the Hive docs pages, so it needs to be dark-mode aware
   - Similarly to the HiveFooter component from previous PRs.

### CODE REVIEW

File: packages/components/src/components/hive-navigation/local-image-for-stories.png
Comment (saihaj) [inline packages/components/src/components/hive-navigation/local-image-for-stories.png:1]: where is this showed?

File: packages/components/src/products.ts [hunk 1]
```diff
@@ -215,3 +212,12 @@ export const PRODUCTS: Record<
     primaryColor: '#000',
   },
 };
+
+export const sixHighlightedProducts = [
```
Comment (saihaj) [inline packages/components/src/products.ts:216]: nit: UPPER_SNAKE_CASE for constants
```suggestion
export const SIX_HIGHLIGHTED_PRODUCTS = [
```

**Review by saihaj — COMMENTED** [review]
left some comments in our slack thread

File: packages/components/src/components/hive-navigation/local-image-for-stories.png
Comment (hasparus) [reply packages/components/src/components/hive-navigation/local-image-for-stories.png:1]: It's imported in hive-navigation stories for the conference card.

File: packages/components/src/products.ts (hunk 1 above)
Comment (hasparus) [reply packages/components/src/products.ts:216]: Good idea.

**Review by saihaj — APPROVED** [review]

**Pull request merged**